src/intersections/rays.jl

```diff
@@ -0,0 +1,61 @@
+# ------------------------------------------------------------------
+# Licensed under the MIT License. See LICENSE in the project root.
+# ------------------------------------------------------------------
+
+#=
+The intersection type can be one of six types:
+1. intersect at one inner point (CrossingRays -> Point)
+2. intersect at origin of one ray (MidTouchingRays -> Point)
+3. intersect at origin of both rays (CornerTouchingRays -> Point)
+4. overlap with aligned vectors (OverlappingAgreeingRays -> Ray)
+5. overlap with colliding vectors (OverlappingOpposingRays -> Segment)
+6. do not overlap nor intersect (NoIntersection)
+=#
+function intersection(f, r1::Ray{N,T}, r2::Ray{N,T}) where {N,T}
```

Comment:
I think we can omit the N and T parameters given that they are not used in the body of the function.

Reply:
`T` is used in the code to decide whether the parameters are approximately zero, so would leave it as it is.